rescript

```diff
@@ -460,17 +462,21 @@ if (
         } else {
           dlog(`Rebuilding since ${reasons_to_rebuild}`);
         }
-        reasons_to_rebuild = [];
-        var p = child_process.spawn(bsb_exe, [], {
-          stdio: ["inherit", "inherit", "pipe"],
-        });
-        p.on("exit", build_finished_callback);
-        p.stderr
-          .setEncoding("utf8")
+
+        child_process
+          .spawn(bsb_exe, [], {
+            stdio: ["inherit", "inherit", "pipe"],
+          })
           // @ts-ignore
           .on("data", function (s) {
             error_output(s, "ninja: error");
-          });
+          })
+          .on("exit", build_finished_callback)
+          .stderr.setEncoding("utf8");
+        // This is important to clean up all 
+        // previous queued events 
+        reasons_to_rebuild = [];
+        LAST_BUILD_START = +Date.now();
```

Comment:
every time we start a build, we log `LAST_BUILD_START`
If we noticed a new file event which is very close to `LAST_BUILD_START`, it is ignored.
5ms delay seems to be okay, since after `LAST_BUILD_START`, it will go through a bunch of code to trigger ninja (which should be more than 5ms consider two processes spawned) so the new file change(even if it is there) will be noticed by ninja.
This change is not for perf benefits, it is mostly used to avoid spamming events(when lots of file changes happened or some buggy file watcher), perf is its side effect.

Reply:
So instead of postponing the new build, it never happens correct?. With the idea that it does not need to happen, because of the 5ms latency and ninja.
The assumption is that these context properties about ninja will not change in future.

Reply:
Note there are two kinds of events spamming.
In vscode, there is a bug that  writing to a file, two file changed events will be emitted.
For truly multiple events, 5ms estimation is a conservative approximation for current model, since the `LAST_BUILD_START` just logged the time when the build is put into event loop, it takes time for it to be executed, where two sequential processes get spawned: rescript.exe and ninja.exe thereafter. Spawning a very lightweight process on my  machine takes 4ms. we can change it to 3ms if that makes sense